### PR TITLE
[RJSF] Menu-component portal shift for tooltips and add slider for percent

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
Signed-off-by: Abhishek-kumar09 <abhimait1909@gmail.com>

**Description**
1. Fix Menu component not showing and clicking on it closes the tooltip
2. Add slider for the percent fields
image

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/MesheryMeshInterface/helpers.js [hunk 1]
```diff
@@ -298,6 +308,7 @@ export function buildUiSchema(schema) {
   const uiSchemaObj = {};
   // 1. Build ui schema
   jsonSchemaBuilder(schema, uiSchemaObj);
+  console.log("ioSchema", schema)
```
Comment (Abhishek-kumar09) [inline ui/components/MesheryMeshInterface/helpers.js:311]:
```suggestion
```

**Comment by leecalcote** [comment]
This change is universal, right? Any field of type “percent” will be affected, yes?

**Comment by Abhishek-kumar09** [comment]
Yes, if the label for text-field or the key of the text-field is including the `percent` as a keyword, it will show the range slider. 
But at very few places the label of the textfield is not marked as percent, eg. is:
image
here the the key/label of the text-field is `value` and not percentage, that's why the slider is not present.

**Comment by Abhishek-kumar09** [comment]
> This change is universal, right?

Yes, The change will be reflected in the extension as well as meshery

File: ui/components/MesheryMeshInterface/helpers.js [hunk 2]
```diff
@@ -265,6 +265,13 @@ function jsonSchemaBuilder(schema, obj) {
   if (schema.type === 'object') {
     for (let key in schema.properties) {
       obj[key] = {};
+
+      // handle percentage for range widget
+      if ((schema.properties?.[key]["type"] === 'number' || schema.properties?.[key].type === 'integer')
```
Comment (Chadha93) [inline ui/components/MesheryMeshInterface/helpers.js:270]: @Abhishek-kumar09 Will this also work for deep fields in the schema(Additional Properties)?

File: ui/components/MesheryMeshInterface/PatternService/RJSF.js [hunk 3]
```diff
@@ -87,6 +87,14 @@ function RJSFForm(props) {
     ObjectFieldTemplate = MesheryCustomObjFieldTemplate,
   } = props;
 
+  useEffect(() => {
+    const extensionTooltipPortal = document.getElementById("meshmap-tooltip-portal");
+    if (extensionTooltipPortal) {
+      rjsfTheme.props.MuiMenu.container = extensionTooltipPortal;
+    }
+    rjsfTheme.zIndex.modal = 99999;
```
Comment (leecalcote) [inline ui/components/MesheryMeshInterface/PatternService/RJSF.js:95]: Zcalc()? Not used in Meshery UI, right?

Comment (Abhishek-kumar09) [reply ui/components/MesheryMeshInterface/PatternService/RJSF.js:95]: NO, not right now.

File: ui/components/MesheryMeshInterface/helpers.js (hunk 2 above)
Comment (Abhishek-kumar09) [reply ui/components/MesheryMeshInterface/helpers.js:270]: It works by traversing any level deep field that comes as rjsf schema. Can't ensure about additional properties though. If the additional properties are the part of schema coming from backend, then Yes.

Comment (leecalcote) [reply ui/components/MesheryMeshInterface/helpers.js:270]: Rephrased: "Yes, if present". Correct?

File: ui/components/MesheryMeshInterface/PatternService/RJSF.js [hunk 4]
```diff
@@ -87,6 +87,14 @@ function RJSFForm(props) {
     ObjectFieldTemplate = MesheryCustomObjFieldTemplate,
   } = props;
 
+  useEffect(() => {
+    const extensionTooltipPortal = document.getElementById("meshmap-tooltip-portal");
```
Comment (leecalcote) [inline ui/components/MesheryMeshInterface/PatternService/RJSF.js:91]: @Abhishek-kumar09 please use a generic name for to reference this extension point / extension.

**Review by leecalcote — APPROVED** [review]

File: ui/components/MesheryMeshInterface/helpers.js (hunk 2 above)
Comment (Abhishek-kumar09) [reply ui/components/MesheryMeshInterface/helpers.js:270]: yes

**Pull request merged**